app/src/main/java/com/infomaniak/drive/ui/addFiles/ImportFilesDialog.kt

```diff
@@ -117,27 +119,22 @@ class ImportFilesDialog : DialogFragment() {
                 val fileName = cursor.getFileName(uri)
                 val (fileCreatedAt, fileModifiedAt) = getFileDates(cursor)
 
-                when {
-                    isLowMemory() -> withContext(Dispatchers.Main) {
-                        showSnackbar(R.string.uploadOutOfMemoryError, showAboveFab = true)
-                    }
-                    else -> {
-                        val outputFile = getOutputFile(uri, fileModifiedAt)
-                        ensureActive()
-                        UploadFile(
-                            uri = outputFile.toUri().toString(),
-                            driveId = navArgs.driveId,
-                            fileCreatedAt = fileCreatedAt,
-                            fileModifiedAt = fileModifiedAt,
-                            fileName = fileName,
-                            fileSize = outputFile.length(),
-                            remoteFolder = navArgs.folderId,
-                            type = UploadFile.Type.UPLOAD.name,
-                            userId = AccountUtils.currentUserId,
-                        ).store()
-                        successCount++
-                        currentImportFile = null
-                    }
+                if (!isLowMemory()) {
```

Comment:
We could leave early using the following code instead of having to write the `if` as is

``` kt
if (isLowMemory()) return@withContext
```

And we can also avoid computing the file name and the files dates if are not going to use them anyway because memory is low